### DISCUSSION TITLE
Prevent `POST` request to be invoked twice (server-side and client-side)

### DISCUSSION THREAD

**Discussion by henningko** [nuxt/nuxt#16445] [Questions] [answered]
Hello,

I am using a Supabase Edge Function that populates an entry in my database. The function is invoked on a protected page (e.g. `settings/auth-verify`) in the `script setup`:
```js
// pages/settings/auth-verify/index.vue
definePageMeta({
  middleware: "auth",
  layout: "app",
});

const { data, error } = await client.functions.invoke("auth-verify", {
  body: JSON.stringify({
    state: route.query.state,
    code: route.query.code,
    scope: route.query.scope,
  }),
});
```

This page is called from a redirect in an auth flow. This results in the function being invoked twice, once from SSR and once on the client side (the latter causing an error, since the auth code is already used).

What would be a good strategy to ensure the function is only invoked once, or best react to being called twice?

Thanks!

**Comment by henningko** [comment]
For the time being, I wrapped it into a `if (process.client) { ... }`.

**Comment by wobsoriano** [comment, marked as answer]
Wrap it in a useAsyncData and set the server option to false 😅

**Reply by henningko** [reply]
Too easy 😂
